cclib/parser/gaussianparser.py

```diff
@@ -1554,6 +1554,74 @@ def extract(self, inputfile, line):
                 self.nocoeffs = nocoeffs
                 self.nooccnos = nooccnos
 
+
+        # Natural spin orbital coefficients (nsocoeffs) and occupation numbers (nsooccnos)
+        if line[5:39] == "Alpha Natural Orbital Coefficients" or line[5:38] == "Beta Natural Orbital Coefficients":
+            if line[5:38] == "Beta Natural Orbital Coefficients":
+                beta = True
+                self.nsocoeffs.append(numpy.zeros((self.nmo, self.nbasis), "d"))
+                self.nsooccnos.append(list())
+            else:
+              beta = False
+              self.aonames = []
+              self.atombasis = []
+              nsocoeffs = [numpy.zeros((self.nmo, self.nbasis), "d")]
+              nsooccnos = [list()]
+            base = 0
+            self.popregular = False
+            for base in range(0, self.nmo, 5):
+                self.updateprogress(inputfile, "Natural Spin orbitals", self.fupdate)
+                colmNames = next(inputfile)
+                if base == 0 and int(colmNames.split()[0]) != 1:
+                    # Implies that this is a POP=REGULAR calculation
+                    # and so, only aonames (not mocoeffs) will be extracted
+                    self.popregular = True
+
+                eigenvalues = next(inputfile)
+                if beta == False:
+                  nsooccnos[0].extend(map(float, eigenvalues.split()[2:]))
+                else:
+                  self.nsooccnos[1].extend(map(float, eigenvalues.split()[2:]))
```

Comment:
I find this part a bit odd (along with similar below). Why are you using `nsooccnos` for alpha but `self.nsooccnos` for beta? Is there a better way to do this?